memestra/memestra.py

```diff
@@ -198,22 +200,25 @@ def collect_deprecated(self, node, duc, ancestors):
                     self.extract_decorator_from_parents(
                         ancestors.parents(user.node),
                         deprecated)
-
         return deprecated
 
     def extract_decorator_from_parents(self, parents, deprecated):
         parent = parents[-1]
         if isinstance(parent, _defs):
-            deprecated.add(parent)
+            deprecated.add(make_deprecated(parent))
             return
         if len(parents) == 1:
             return
         parent_p = parents[-2]
         if isinstance(parent, ast.Call) and isinstance(parent_p, _defs):
-            deprecated.add(parent_p)
+            reason = None
+            # Output only the specified reason with the --reason-keyword flag
+            for keyword in parent.keywords:
+                if self.reason_keyword == keyword.arg:
+                    reason = keyword.value.value
```

Comment:
critical: you should only do this if the reason is a string literal. Otherwise warn the user somehow.

Reply:
Right!